extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java

```diff
@@ -174,4 +180,48 @@
         public Optional<List<String>> includePatterns;
     }
 
+    @ConfigGroup
+    public static class RuntimeCatalogConfig {
+        /**
+         * Enable {@link CamelRuntimeCatalog} functionaries.
+         */
+        @ConfigItem(defaultValue = "true")
+        public boolean enabled;
+
+        /**
+         * Used to control the resolution of components catalog info.
```

Comment:
Is "resolution" really what this does? Would not "inclusion" suit better?

Again, it may be unclear to the reader, what is catalog. Maybe a link to a Camel docs page explaining the concept would be nice. The description should give me a hint why/when should I care, why would I ever need this to be true?

This also applies to all other options in this file.

Reply:
> Is "resolution" really what this does? Would not "inclusion" suit better?
> 


It is resolution, if you go through the code you'd see that this flag controls if the runtime catalog should attempt or not to resolve the given name to its model


> Again, it may be unclear to the reader, what is catalog. Maybe a link to a Camel docs page explaining the concept would be nice. The description should give me a hint why/when should I care, why would I ever need this to be true?
> 

The impact on the native image is described a few lines below.

Reply:
should be fixed now